src/app/clusters/wifi-network-diagnostics-server/wifi-network-diagnostics-cluster.h

```diff
@@ -0,0 +1,121 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/server-cluster/DefaultServerCluster.h>
+#include <clusters/WiFiNetworkDiagnostics/ClusterId.h>
+#include <clusters/WiFiNetworkDiagnostics/Enums.h>
+#include <clusters/WiFiNetworkDiagnostics/Metadata.h>
+#include <lib/core/DataModelTypes.h>
+#include <platform/DiagnosticDataProvider.h>
+#include <protocols/interaction_model/StatusCode.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+
+struct WiFiNetworkDiagnosticsEnabledAttributes
+{
+    bool enableBeaconLostCount : 1;
+    bool enableBeaconRxCount : 1;
```

Comment:
This should only have flags for attributes that are optional, not that are feature-controlled.  Right now we're allowing someone to enable a feature but not the then-required attributes, which seems like a footgun we should not have.